### DISCUSSION TITLE
Keycloak 17 Admin Console Redirecting To Port 443

### DISCUSSION THREAD

**Discussion by soundsessential** [keycloak/keycloak#10331] [Archive]
I've been trying to figure out Keycloak 17 and certainly making progress. I've installed a MySQL database, downloaded Lets Encrypt certs and configured all this in the conf file. After build, adding the initial admin user and then start, Keycloak is available on:
https://keycloak.server.com:8443

That lands on the Welcome to Keycloak page. If I click on the Admin Console link to log into Keycloak, the address changes from using the port number to:
https://keycloak.server.com/admin/

Which fails because it's not set up to be on port 443. Even if I add the port to the above address, it strips it and goes back to straight https. If I change the port to 443 in the conf file, then I can access the initial page on:
https://keycloak.server.com

And then clicking the admin console link does work because it's all on 443. 

But I want to run Keycloak on 8443. Why's it not keeping the port number in the address?

I'll check in a bit, but I'm pretty certain this was not the case in 16. Many thanks for any pointers.

**Comment by soundsessential** [comment]
And the answer is, use this in the conf file:
`# Hostname for the Keycloak server.
hostname=keycloak.server.com:8443`

**Comment by stianst** [comment]
Are you enabling proxy mode (https://www.keycloak.org/server/all-config#_proxy)? If not it should not use the 443 port, so that would be a bug. You should not need to include port in the `--hostname` option, in fact I'm surprised that even works as it shouldn't.

**Reply by soundsessential** [reply]
Many thanks for the response. Proxy mode was not enabled in the situation above.

**Comment by stianst** [comment]
Looks like this issue https://github.com/keycloak/keycloak/issues/10627

**Reply by soundsessential** [reply]
Cool, I'll follow that and see what happens with it. Again, much appreciated.

**Comment by raz08** [comment]
@soundsessential , Is it working ? I have the same issue where when I click on admin console it is taking to default port 443 but that is exposed outside. 

Please let me know if you resolved it. Share the config. Thanks

**Reply by soundsessential** [reply]
I initially resolved it by adding the port number in the host name as above. But since then I think some code changes have taken place. I've not done an install for a while now but what version are you using?

**Comment by raz08** [comment]
19.0.3

**Reply by soundsessential** [reply]
Yeah I've not used that version yet so not sure I can help. Sorry. Try following the links above though, see if they provide any clues.

**Reply by splatch** [reply]
It looks like 19.0.2 is having similar issue with some uris, for me I get `auth/realms/master/protocol/openid-connect/3p-cookies/step1.html` stuck to https while other parts are fine with http.
